dynamicArray.h

```diff
@@ -0,0 +1,100 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define INIT_SIZE 1
+#define EXPAND_COEFFICIENT 2
+
+typedef enum {
+    SUCCESS = 0,
+    INIT_ERROR,
+    EXPAND_ERROR,
+    SHRINK_ERROR,
+    INDEX_ERROR,
+} DynamicArrayError;
+
+typedef struct {
+    int* head;
+    size_t current_size;
+    size_t max_capacity;
+} DynamicArray;
+
+/* 
+Initialize dynamic array.
+
+@param arr - pointer to the initialized array.
+
+@return DynamicArrayError enum with exit code.
+*/
+DynamicArrayError initArray(DynamicArray* arr);
+
+/* 
+Free dynamic array.
+
+@param arr - the array to free. 
+*/
+void freeArray(DynamicArray* arr);
+
+/*
+Expand the given array by factor of the defined EXPAND_COEFFICIENT.
+
+@param arr - the array to expand. 
+
+@return DynamicArrayError enum with exit code.
+*/
+DynamicArrayError expandArray(DynamicArray* arr);
+
+/* 
+Shrink the given array by factor of the define EXPAND_COEFFICIENT. 
+
+@param arr - the array to shrink.
+
+@return DynamicArrayError enum with exit code.
+*/
+DynamicArrayError shrinkArray(DynamicArray* arr);
```

Comment:
Should these functions be in the header file? Do you want the external user to be able to arbitrarily expand and shrink your dynamic array?